pkg/virt-controller/services/net_binding_test.go

```diff
@@ -103,28 +107,34 @@ var _ = Describe("Network Binding", func() {
 		})
 	})
 
-	Context("binding plugin configuration", func() {
-		It("should read registered network binding plugins successfully", func() {
-			const testPluginName = "test"
-			testPlugin := v1.InterfaceBindingPlugin{SidecarImage: "kubevirt/network-slirp-binding"}
-
+	Context("network binding plugin configuration", func() {
+		It("should create Slirp hook sidecar with registered image (specified in Kubevirt config)", func() {
+			fakeRecorder := record.NewFakeRecorder(1)
+			vmi := v1.NewVMI("test", "1234")
 			config := &v1.KubeVirtConfiguration{
+				ImagePullPolicy: k8scorev1.PullIfNotPresent,
 				NetworkConfiguration: &v1.NetworkConfiguration{
 					Binding: map[string]v1.InterfaceBindingPlugin{
-						"plugin1":      {SidecarImage: "org1/plugin1"},
-						testPluginName: testPlugin,
-						"plugin2":      {SidecarImage: "org2/plugin2"},
+						"slirp": {SidecarImage: "kubevirt/network-slirp-plugin"},
 					},
 				},
 			}
 
-			Expect(services.ReadNetBindingPluginConfiguration(config, testPluginName)).To(Equal(&testPlugin))
+			Expect(services.SlirpNetBindingPluginSidecar(vmi, config, fakeRecorder)).To(Equal(hooks.HookSidecar{
+				ImagePullPolicy: k8scorev1.PullIfNotPresent,
+				Image:           "kubevirt/network-slirp-plugin",
+			}))
 		})
 
-		DescribeTable("should return no network binding plugins configuration, when config", func(config *v1.KubeVirtConfiguration) {
-			Expect(services.ReadNetBindingPluginConfiguration(config, "myplugin")).To(BeNil())
-		},
-			Entry("is nil", nil),
+		DescribeTable("should create Slirp hook sidecar with default image, when Kubevirt config",
+			func(config *v1.KubeVirtConfiguration) {
+				fakeRecorder := record.NewFakeRecorder(1)
```

Comment:
I think you are suppose to check that the event was emitted by examining `fakeRecorder.Events`.

Reply:
Do we check warning events emitting in unit tests?

Reply:
You have a flake here, you are suppose to check what it recorded. That is why a fake was created, to allow such checks.

Reply:
A fake was created just to satisfy `services.SlirpNetBindingPluginSidecar`, this tests are about checking slirp default image is set.
No flakes whatsoever.

I added separate test to verify an event is raised.